webapp/src/hooks/useForm.ts

```diff
@@ -0,0 +1,121 @@
+import {useState} from 'react';
+
+// Set initial value of form fields
+const getInitialFieldsValue = (
+    formFields: Record<FormFields, ModalFormFieldConfig>,
+): Record<FormFields, string> => {
+    let fields = {};
+    Object.keys(formFields).forEach((field) => {
+        fields = {
+            ...fields,
+            [field as FormFields]:
+                formFields[field as FormFields].value ||
+                (field as FormFields === 'timestamp' ? Date.now().toString() : ''),
+        };
+    });
+
+    return fields as unknown as Record<FormFields, string>;
+};
+
+/**
+ * Filter out all the fields for which validations check required
+ * and set empty string as default error message
+ */
+const getFieldslWhereErrorCheckRequired = (
+    formFields: Record<FormFields, ModalFormFieldConfig>,
+): Partial<Record<FormFields, string>> => {
+    let fields = {};
+    Object.keys(formFields).forEach((field) => {
+        if (formFields[field as FormFields].validations) {
+            fields = {
+                ...fields,
+                [field as FormFields]: '',
+            };
+        }
+    });
+
+    return fields as unknown as Partial<Record<FormFields, string>>;
+};
+
+// Check each type of validations and return required error message
+const getValidationErrorMessage = (
+    formFields: Record<FormFields, string>,
+    fieldName: FormFields,
+    fieldLabel: string,
+    validationType: ValidationTypes,
+): string => {
+    switch (validationType) {
+    case 'isRequired':
+        return formFields[fieldName] ? '' : `${fieldLabel} is required`;
+    default:
+        return '';
+    }
+};
+
+// Genric hook to handle form fields
+function useForm(initialFormFields: Record<FormFields, ModalFormFieldConfig>) {
+    // Form field values
+    const [formFields, setFormFields] = useState(getInitialFieldsValue(initialFormFields));
+
+    // Form field error state
+    const [errorState, setErrorState] = useState<Partial<Record<FormFields, string>>>(
+        getFieldslWhereErrorCheckRequired(initialFormFields),
+    );
+
+    /**
+     * Set new field value on change
+     * and reset field error state
+     */
+    const onChangeOfFormField = (fieldName: FormFields, value: string) => {
+        setErrorState({...errorState, [fieldName]: ''});
+        setFormFields({...formFields, [fieldName]: value});
+    };
+
+    // Validate all form fields and set error if any
+    const isErrorInFormValidation = (): boolean => {
+        let fields = {};
```

Comment:
Rename this variable to something else to denote that it is for storing error states.